### PR TITLE
`Development`: Restructure CompetencyIntegrationTest

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Description

Utilizes nested test classes to improve the code quality of tests in `CompetencyIntegrationTest`.

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unchanged

## Summary by CodeRabbit

- **Tests**
    - Updated the architecture of test classes for improved clarity and control flow.
    - Renamed nested classes in service tests for consistency and readability.

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/lecture/CompetencyIntegrationTest.java [hunk 1]
```diff
@@ -720,19 +796,22 @@ void addPrerequisite() throws Exception {
...
 
-    @Test
-    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
-    void removePrerequisite_bad_request() throws Exception {
-        request.delete("/api/courses/" + course.getId() + "/prerequisites/" + competency.getId(), HttpStatus.BAD_REQUEST);
+        @Test
+        @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+        void shouldReturnBadRequestWhenPrerequisiteNotExists() throws Exception {
+            request.delete("/api/courses/" + course.getId() + "/prerequisites/" + competency.getId(), HttpStatus.BAD_REQUEST);
+        }
```
[the remark this answers is not in the document]
Comment (MaximilianAnzinger) [reply src/test/java/de/tum/in/www1/artemis/lecture/CompetencyIntegrationTest.java:814]: Yes, please.

**Review by JohannesWt — APPROVED** [review]
Code looks good

File: src/test/java/de/tum/in/www1/artemis/lecture/CompetencyIntegrationTest.java [hunk 2]
```diff
@@ -593,84 +659,113 @@ void getCompetencyStudentProgressTest() throws Exception {
...
+        @Test
+        @WithMockUser(username = TEST_PREFIX + "instructor42", roles = "INSTRUCTOR")
+        void shouldReturnForbiddenForInstructorOfOtherCourse() throws Exception {
+            Competency competency = new Competency();
+            competency.setTitle("Example Title");
+            request.postWithResponseBody("/api/courses/" + course2.getId() + "/competencies/import", competency, Competency.class, HttpStatus.FORBIDDEN);
+        }
     }
 
     @Test
```
Comment (rstief) [inline src/test/java/de/tum/in/www1/artemis/lecture/CompetencyIntegrationTest.java:771]: could also be added to own test class

**Review by rstief — COMMENTED** [review]
Mainly notes on test naming conventions

**Review by JohannesWt — APPROVED** [review]

**Review by Jan-Thurner — DISMISSED** [review]
Code LGTM

**Comment by krusche** [comment]
@MaximilianAnzinger please resolve the merge conflict

**Pull request merged**